.github/workflows/odh-notebooks-sync.yml

```diff
@@ -0,0 +1,132 @@
+---
+# The aim of this GitHub workflow is to update the pipfile to sync with Codeflare-SDK release.
+name: Sync with codeflare-sdk release
+on:  # yamllint disable-line rule:truthy
+  workflow_dispatch:
+    inputs:
+      upstream-repository-organization:
+        required: true
+        description: "Provide the upstream repository organization name"
+
+      codeflare-repository-organization:
+        required: true
+        description: "Provide the Codeflare repository organization name"
+
+      codeflare_sdk_release_version:
+        required: true
+        description: "Provide version of the Codeflare-SDK release"
+
+env:
+  BRANCH_NAME: main
+  CODEFLARE_RELEASE_VERSION: ${{ github.event.inputs.codeflare_sdk_release_version }}
+  UPDATER_BRANCH: odh-sync-updater-${{ github.run_id }}
+  UPSTREAM_OWNER: ${{ github.event.inputs.upstream-repository-organization }}
+  REPO_OWNER: ${{ github.event.inputs.codeflare-repository-organization }}
+  REPO_NAME: notebooks
+  GITHUB_TOKEN: ${{ secrets.TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ env.BRANCH_NAME }}
+
+      - name: Clone repository and Sync
+        run: |
+          git clone https://x-access-token:${GITHUB_TOKEN}@github.com/$REPO_OWNER/$REPO_NAME.git $REPO_NAME
+          cd $REPO_NAME
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/$REPO_OWNER/$REPO_NAME.git
+          git remote add upstream https://github.com/$UPSTREAM_OWNER/$REPO_NAME.git
+          git remote -v
+          git fetch upstream
```

Comment:
IMHO instead of fetch and merge you can leverage `git pull upstream main && git push origin main` as done in https://github.com/project-codeflare/codeflare-operator/blob/main/Makefile#L305

Reply:
that helps you to avoid merge commits

Reply:
Thank you @sutaakar for reviewing,  I have made changes accordingly !!